### PR TITLE
Feature/ker/qiskit simulator

### PR DESCRIPTION
**Pull request by keriksson-rosenqvist**
Noise aware qiskit simulation backend with some additional improvements pulled in from develop (aka `model.create_xyz`)

### CODE REVIEW

File: src/QAT/qat/purr/backends/qiskit_simulator.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        return QiskitRuntime(existing_engine or self.create_engine())
+
+    def create_builder(self) -> InstructionBuilder:
+        return QiskitBuilder(hardware_model=self, qubit_count=self.qubit_count)
+
+    def create_engine(self) -> InstructionExecutionEngine:
+        return QiskitExecutor(hardware_model=self)
+
+
+class QiskitExecutor(InstructionExecutionEngine):
```
Comment (chemix-lunacy) [inline src/QAT/qat/purr/backends/qiskit_simulator.py:173]: `QiskitEngine`

File: src/QAT/qat/purr/backends/realtime_chip_simulator.py [hunk 2]
```diff
@@ -644,21 +644,13 @@ def get_default_RTCS_hardware(repeats=1000, rotating_frame=True):
...
-        builder_types: List[QuantumInstructionBuilder] = None,
-        **kwargs,
-    ):
-        super().__init__(
-            *args,
-            engine_types=engine_types or [RealtimeChipSimEngine],
-            builder_types=builder_types or [QuantumInstructionBuilder],
-            **kwargs,
-        )
+    def __init__(self, *args, **kwargs):
```
Comment (chemix-lunacy) [inline src/QAT/qat/purr/backends/realtime_chip_simulator.py:647]: This dosen't need to exist at all if you're not doing anything in the constructor.

File: src/QAT/qat/purr/compiler/builders.py [hunk 3]
```diff
@@ -45,6 +46,12 @@
 log = get_default_logger()
 
 
+class Axis(Enum):
```
Comment (chemix-lunacy) [inline src/QAT/qat/purr/compiler/builders.py:49]: Not used anywhere I can see?

File: src/QAT/qat/purr/backends/verification.py [hunk 4]
```diff
@@ -155,5 +157,8 @@ def verify_program(
     return execute(program, model, compiler_config)
 
 
+verification_engines: Dict[str, VerificationEngine] = {"Lucy": LucyVerificationEngine}
```
Comment (chemix-lunacy) [inline src/QAT/qat/purr/backends/verification.py:160]: If this sticks around it should use develops approach instead: https://github.com/oqc-community/qat/pull/48/files

**Review by chemix-lunacy — CHANGES_REQUESTED** [review]
You shouldn't be porting back the various model changes - this branch should only have things that are absolutely critical going into it. No improvements that aren't directly required.

Also I'd like to see this go into develop before I approve here.

File: src/QAT/qat/purr/compiler/builders.py (hunk 3 above)
Comment (keriksson-rosenqvist) [reply src/QAT/qat/purr/compiler/builders.py:49]: https://github.com/oqc-community/qat/pull/49/files#diff-ce42f535a52f24404f7a982ad24a8b40e2057434b056ffc9c56e32625ddc1e16R54

**Comment by keriksson-rosenqvist** [comment]
Develop merged

**Review by jfriel-oqc — APPROVED** [review]

**Review by chemix-lunacy — APPROVED** [review]

**Pull request merged**